### PR TITLE
Remove JSONB support

### PR DESCRIPTION
**Pull request by lutter**
The PR we've all been waiting for: remove support for JSONB storage and simplify the code because we do not need to accommodate two different storage schemes.

Note that this PR applies on top of #1992

### CODE REVIEW

File: store/postgres/src/entities.rs [hunk 1]
```diff
@@ -666,7 +456,7 @@ impl Connection<'_> {
         // 5 minutes
         const MIGRATION_CHECK_FREQ: u64 = 20;
 
-        if self.storage.needs_migrating() {
+        if false {
```
Comment (leoyvens) [inline store/postgres/src/entities.rs:459]: `if false`?

File: store/postgres/src/entities.rs [hunk 2]
```diff
@@ -700,7 +490,7 @@ impl Connection<'_> {
     ) -> Result<bool, Error> {
```
Comment (leoyvens) [inline store/postgres/src/entities.rs:490]: The return type can be changed to `()` I believe.

**Review by leoyvens — APPROVED** [review]
This is the most beautiful PR I've ever reviewed!

File: store/postgres/src/entities.rs (hunk 1 above)
Comment (lutter) [reply store/postgres/src/entities.rs:459]: I didn't quite want to do away with the code for subgraph migrations when I wrote this; but commit 6e5424f3 in `lutter/shard` will do away with this nonsense

File: store/postgres/src/entities.rs (hunk 2 above)
Comment (lutter) [reply store/postgres/src/entities.rs:490]: Likewise, that same commit will unceremoniously yank all that code.

**Comment by lutter** [comment]
> This is the most beautiful PR I've ever reviewed!

Haha .. yes, nothing better than deleting 2k lines of code

**Pull request merged**